src/modules/Lan.rb

```diff
@@ -348,38 +348,47 @@ def Read(cache)
       return false if Abort()
       # Progress step 4/8
       ProgressNextStage(_("Reading network configuration...")) if @gui
-      NetworkConfig.Read
+      begin
+        NetworkConfig.Read
 
-      @ipv6 = readIPv6
+        @ipv6 = readIPv6
 
-      Builtins.sleep(sl)
+        Builtins.sleep(sl)
 
-      return false if Abort()
-      # Progress step 5/8
-      ProgressNextStage(_("Reading hostname and DNS configuration...")) if @gui
-      DNS.Read
-      Host.Read
-      Builtins.sleep(sl)
+        return false if Abort()
+        # Progress step 5/8
+        ProgressNextStage(_("Reading hostname and DNS configuration...")) if @gui
+        DNS.Read
 
-      return false if Abort()
-      # Progress step 6/8
-      ProgressNextStage(_("Reading installation information...")) if @gui
-      #    ReadInstallInf();
-      Builtins.sleep(sl)
+        Host.Read
+        Builtins.sleep(sl)
 
-      return false if Abort()
-      # Progress step 7/8
-      ProgressNextStage(_("Reading routing configuration...")) if @gui
-      Routing.Read
-      Builtins.sleep(sl)
+        return false if Abort()
+        # Progress step 6/8
+        ProgressNextStage(_("Reading installation information...")) if @gui
+        Builtins.sleep(sl)
 
-      return false if Abort()
-      # Progress step 8/8
-      ProgressNextStage(_("Detecting current status...")) if @gui
-      NetworkService.Read
-      Builtins.sleep(sl)
+        return false if Abort()
+        # Progress step 7/8
+        ProgressNextStage(_("Reading routing configuration...")) if @gui
+        Routing.Read
+        Builtins.sleep(sl)
+
+        return false if Abort()
+        # Progress step 8/8
+        ProgressNextStage(_("Detecting current status...")) if @gui
+        NetworkService.Read
+        Builtins.sleep(sl)
+
+        return false if Abort()
+      rescue IOError, SystemCallError, RuntimeError => error
+        msg = format(_("Network configuration is corrupted.\n"\
+                "If you continue resulting configuration can be malformed."\
+                "\n\n%s"), wrap_text(error.message))
+        ret = @gui && Popup.ContinueCancel(msg)
+        return false if !ret
```

Comment:
I find this logic hard to understand, I would write
```rb
  return false unless @gui
  return false unless ContinueCancel...
```
but it's your call.

Reply:
ok